test/runtests.jl

```diff
@@ -1,33 +1,70 @@
-using Distributed, Test, CUDA
+using Distributed, Test, CUDA, Pkg
+
+using AdvancedHMC: AdvancedHMC
 
 println("Envronment variables for testing")
 println(ENV)
 
+const DIRECTORY_AdvancedHMC = dirname(dirname(pathof(AdvancedHMC)))
+const DIRECTORY_Turing_tests = joinpath(DIRECTORY_AdvancedHMC, "test", "turing")
+const GROUP = get(ENV, "GROUP", "All")
+
 @testset "AdvancedHMC" begin
-    tests = [
-        "metric",
-        "hamiltonian",
-        "integrator",
-        "trajectory",
-        "adaptation",
-        "sampler",
-        "sampler-vec",
-        "demo",
-        "models",
-    ]
-
-    if CUDA.functional()
-        @eval module TestCUDA
+    if GROUP == "All" || GROUP == "AdvancedHMC"
+        tests = [
+            "metric",
+            "hamiltonian",
+            "integrator",
+            "trajectory",
+            "adaptation",
+            "sampler",
+            "sampler-vec",
+            "demo",
+            "models",
+        ]
+
+        if CUDA.functional()
+            @eval module TestCUDA
             include("cuda.jl")
+            end
+        else
+            @warn "Skipping GPU tests because no GPU available."
         end
-    else
-        @warn "Skipping GPU tests because no GPU available."
-    end
 
-    res = map(tests) do t
-        @eval module $(Symbol("Test_", t))
+        res = map(tests) do t
+            @eval module $(Symbol("Test_", t))
             include($t * ".jl")
+            end
+            return
+        end
+    end
+
+    if GROUP == "All" || GROUP == "Downstream"
+        @testset "turing" begin
+            try
```

Comment:
I suppose this block is to handle the change of how test-only dependencies are handled in `Project.toml`?

Reply:
Not sure I completely understand what you mean. But the reason why I introduced it is because there are some namespaces issues, e.g. if `using AdvancedHMC` occurs in any of the tests above, then `using Turing` which happens in `test/turing/runtests.jl` will cause a namespace clash, e.g. `Turing.NUTS` and `AdvancedHMC.NUTS`. This in turn leads to errors in the downstream tests since they expect `NUTS` to be available in the namespace + it refers to `Turing.NUTS`. Putting these in a module avoids this.

Reply:
The namespace issue is handled from Line 58 right?
I'm referring to the codes before that.

Reply:
Oh you're referring to the `try` block? It's to handle the case where we bump the minor or major version of AHMC, which then won't be compatible with Turing. In those cases we just have to go "np, it's a breaking change, so we're going to make changes to Turing anyways".